app/src/main/java/org/mozilla/focus/utils/IntentUtils.java

```diff
@@ -154,9 +154,13 @@ public static void intentOpenSettings(Activity activity, int requestCode) {
     }
 
     public static void intentOpenSettings(Fragment fragment, int requestCode) {
+        Activity host = fragment.getActivity();
+        if (host == null) {
+           return;
+        }
         Intent intent = new Intent();
         intent.setAction(android.provider.Settings.ACTION_APPLICATION_DETAILS_SETTINGS);
-        Uri uri = Uri.fromParts("package", fragment.getActivity().getPackageName(), null);
+        Uri uri = Uri.fromParts("package", host.getPackageName(), null);
         intent.setData(uri);
         fragment.startActivityForResult(intent, requestCode);
```

Comment:
What if the activity is going to be finished, would it be safer if we use activity.startActivityForResult()?
Or, should we check isDestroyed or isFinishing?

Reply:
activity.startActivityForResult() will not work. activity.onActivityResult will be triggered rather then fragment.onActivityResult and that will make permissionhandler hard to use since it will couple the host activity and the fragment.

On the other hand, yeah it never hurts to add some checks but I think that Snackbar 3 seconds is the most crucial part. Lifecycle is like 1/1000 of the reproduce rate.